### PR TITLE
governance: add Subproject Leads as a named role in the SIG governance

### PR DESCRIPTION
**Pull request by palnabarun**
Broadly this PR has the following changes:
- Add Subproject Lead section to SIG governance
- Pare down the responsibilities of the Subproject Owners
- Add field in sigs.yaml to define Subproject Leads and update generator to allow for that
- Point community-membership guideline to governance doc instead of having duplicated content

Steering Votes:

- [x] @BenTheElder
- [x] @cblecker
- [x] @cpanato
- [ ] @justaugustus
- [x] @mrbobbytables
- [x] @palnabarun (author)
- [x] @tpepper

/hold for comments
/committee steering

### CODE REVIEW

**Comment by mrbobbytables** [comment]
Thanks @palnabarun !
Happy to see this happen :)
LGTM from me with steering hat on

**Comment by cpanato** [comment]
lgtm from me with steering hat on as well
 looks nice

**Comment by MadhavJivrajani** [comment]
Super happy to see this happen, thank you @palnabarun  ❤️

**Comment by tpepper** [comment]
This feels to me like a useful expansion and clarification of roles and responsibilities, effectively splitting out a (possibly additional) layer in the contributor ladder.  It's something our chairs/leads have asked for.  For some SIGs/subprojects it's a no-op or a minor swizzle, for some it opens up opportunity and recognizes those who are keeping the lights on in a broad way.

/lgtm
(Steering)

**Comment by neolit123** [comment]
FTR, sig cluster-lifecycle has [24 subprojects ](https://github.com/kubernetes/community/blob/master/sig-cluster-lifecycle/README.md) currently and we have not seen a request for such a distinctive role to exist. the most common ask is the creation of a subproject slack channel.

when a new reviewer/approver is added they must be PRed in the repo OWNERS file. OWNERS files are viewable in the subproject list of the group. if the person also needs a github repo admin/member access they must be PRed in the repo config. if they also wish to be added as a subproject lead, they must be PRed in sigs.yaml of k/community.

this means 3 PRs.

sigs.yaml is still a monolith and sharding never happened for it. this means each lead addition must be top level approved in k/community.

if we had automation and streamlined management of maintainers, i'd be in favor. :)

**Comment by BenTheElder** [comment]
> FTR, sig cluster-lifecycle has [24 subprojects ](https://github.com/kubernetes/community/blob/master/sig-cluster-lifecycle/README.md)currently and we have not seen a request for such a distinctive role to exist. the most common ask is the creation of a subproject slack channel.

In some cases the list will be the same, this was also true with e.g. SIG Chair/TL split where it commonly at least started as the same list of people.

It still clarifies the role and allows flexibility to not have them 1:1.

I think we at least want better framing than punting to OWNERS files, which are a bespoke technical implementation of approval, versus e.g. SIG TLs which are listed explicitly.

> when a new reviewer/approver is added they must be PRed in the repo OWNERS file. OWNERS files are viewable in the subproject list of the group. if the person also needs a github repo admin/member access they must be PRed in the repo config. if they also wish to be added as a subproject lead, they must be PRed in sigs.yaml of k/community.
>
> this means 3 PRs.

This should happen relatively infrequently and in each case involves granting quite a bit of access to that person, I think it's OK that this takes a few trivial PRs.
In some projects github admin may not be necessary (full automation).

**Review by BenTheElder — COMMENTED** [review]
/lgtm
(Steering)

I do think we should revisit at some point and see how unique the two lists are and consider reframing again, but +1 to enable the "subproject lead" role explicitly inline in SIGs' docs.

**Review by cpanato — APPROVED** [review]
/lgtm
(Steering)

File: committee-steering/governance/sig-governance.md
Comment (cblecker) [inline committee-steering/governance/sig-governance.md:167]:
```suggestion
  - *If no Subproject Leads are defined for the subproject, the Subproject Owners will be responsible for the same duties.*
```

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by mrbobbytables** [comment]
/lgtm
(reapplying)

**Comment by cblecker** [comment]
LGTM

**Comment by palnabarun** [comment]
We have reached a quorum with the following votes:

- [x] @BenTheElder
- [x] @cblecker
- [x] @cpanato
- [ ] @justaugustus
- [x] @mrbobbytables
- [x] @palnabarun (author)
- [x] @tpepper

/hold cancel

As a follow-up, I will send an email to k-dev and leads.

**Pull request merged**
